src/GraphQL.MicrosoftDI.Tests/ScopedAttributeTests.cs

```diff
@@ -55,15 +55,15 @@ public async void ScopedSubscriptionWorks()
         };
         var unscopedSubscriptionResolver = graphType.Fields.Find(nameof(TestClass.UnscopedAsyncSubscription))!.StreamResolver!;
         var scopedAsyncSubscriptionResolver = graphType.Fields.Find(nameof(TestClass.ScopedAsyncSubscription))!.StreamResolver!;
-        (await unscopedSubscriptionResolver.ResolveAsync(context).ConfigureAwait(false)).Subscribe(new SampleObserver("0 1"));
-        (await unscopedSubscriptionResolver.ResolveAsync(context).ConfigureAwait(false)).Subscribe(new SampleObserver("1 2"));
-        (await unscopedSubscriptionResolver.ResolveAsync(context).ConfigureAwait(false)).Subscribe(new SampleObserver("2 3"));
+        (await unscopedSubscriptionResolver.ResolveAsync(context).ConfigureAwait(false)).Subscribe(new SampleObserver("0 1")).Dispose();
```

Comment:
Was it a necessity to work properly or just code style issue?

Reply:
Necessary -- although in actual environment, the subscriptions would have been disposed.

Reply:
Ah, I think this is because `Class1.DisposedCount.ShouldBe` checks. Now scope dies not in `ResolveAsync` but when calling `Dispose` on the returned subscription.

Reply:
Correct.